### PR TITLE
Reveal and select created elements in the navigator

### PR DESCRIPTION
**Pull request by franzbecker**

### CODE REVIEW

File: src/lib/src/component/navigation/navigation.component.ts [hunk 1]
```diff
@@ -71,32 +74,49 @@ export class NavigationComponent implements OnInit {
...
     });
     this.messagingService.subscribe(events.NAVIGATION_SELECT, element => {
       this.uiState.selectedElement = element as WorkspaceElement;
       this.changeDetectorRef.detectChanges();
     });
   }
 
+  handleNavigationCreated(payload: any): void {
+    this.retrieveWorkspaceRoot().then(() => {
+      this.revealElement(payload.path);
```
Comment (gunther-bachmann) [inline src/lib/src/component/navigation/navigation.component.ts]: can payload be typed somehow?

Comment (gunther-bachmann) [reply src/lib/src/component/navigation/navigation.component.ts]: what if retrieval returns undefined?

File: src/lib/src/component/navigation/navigation.component.ts [hunk 2]
```diff
@@ -71,32 +74,49 @@ export class NavigationComponent implements OnInit {
...
 
   collapseAll(): void {
     this.uiState.clearExpanded();
-    this.uiState.setExpanded(this.workspaceRoot.path, true);
+    this.uiState.setExpanded(this.workspace.root.path, true);
+  }
+
+  revealElement(path: string): void {
+    let subpaths = this.workspace.getSubpaths(path);
+    subpaths.forEach(subpath => this.uiState.setExpanded(subpath, true));
```
Comment (gunther-bachmann) [inline src/lib/src/component/navigation/navigation.component.ts:116]: does setting the expanded flag of the last subpath display all the sub element, too?
asked differently: shouldn't the expanded flag be set except on the element itself (last subpath)?

File: src/lib/src/component/navigation/navigation.component.ts (hunk 1 above)
Comment (franzbecker) [reply src/lib/src/component/navigation/navigation.component.ts]: We can't type it as it differs from event to event. In `MessagingService` we defined:
```
   subscribe(type: string, callback: (payload: any) => void): Subscription;
```

Comment (franzbecker) [reply src/lib/src/component/navigation/navigation.component.ts]: Added a cast + test for the undefined case

File: src/lib/src/component/navigation/navigation.component.ts (hunk 2 above)
Comment (franzbecker) [reply src/lib/src/component/navigation/navigation.component.ts:116]: Hm maybe the naming is misleading here... subpath means "part of the path" - not sub element.
So `a/b/c` consists of the subpaths `a/`, `a/b` and `a/b/c`. Don't have a better name yet... path fragment is not good either as I would expect a fragment to be only a part like `b/`

**Pull request merged**